src/Stratis.Bitcoin.Tests/Consensus/ChainedHeaderTreeTest.cs

```diff
@@ -327,6 +334,104 @@ public void ConnectHeaders_SupplyHeaders_ToDownloadArraySizeSameAsNumberOfHeader
             Assert.Equal(headersToDownloadCount, peer2Headers.Count);
         }
 
+        /// <summary>
+        /// Issue 7 @ Create chained header tree component #1321
+        /// We have a chain and someone presents an invalid header.
+        /// After that our chain's last block shouldn't change, it shouldn't have a valid .Next
+        /// and it should throw an exception.
+        /// </summary>
+        [Fact]
+        public void ConnectHeaders_SupplyInvalidHeader_ExistingChainTipShouldNotChange()
+        {
+            TestContext testContext = new TestContextBuilder().WithInitialChain(5).UseCheckpoints(false).Build();
+            ChainedHeaderTree chainedHeaderTree = testContext.ChainedHeaderTree;
+            uint256 peerTipHashBeforeInvalidBlockPresented = testContext.InitialChainTip.HashBlock;
```

Comment:
probably name it `consensusTip` and get ChainedHeader and not the hash there

Also
`peerTipAfterInvalidHeaderPresented` => `consensusTipAfterInvalidHeaderPresented`

and instead of 
`Assert.Equal(peerTipHashBeforeInvalidBlockPresented, peerTipAfterInvalidHeaderPresented.HashBlock);`

put there those 2 headers- their hashes will be checked to be equal (ChainedHeader has override on ==)

Reply:
@noescape00 All done.